runtime/turing/src/xcm_config.rs

```diff
@@ -165,19 +172,106 @@ pub type Barrier = DenyThenTry<
 		// ^^^ Parent and its exec plurality get free execution
 	),
 >;
+
+/// Based on the precedent set by other projects. This will need to be changed.
+pub fn ksm_per_second() -> u128 {
+	CurrencyId::KSM.cent() * 16
+}
+
+/// Assuming ~ $0.50 TUR price.
+pub fn tur_per_second() -> u128 {
+	let tur_equivalent = convert_to_token(CurrencyId::KSM, CurrencyId::Native, ksm_per_second());
+	// Assuming KSM ~ $130.00.
+	tur_equivalent * 260
+}
+
+parameter_types! {
+	pub TurPerSecond: (AssetId, u128) = (
+		MultiLocation::new(
+			1,
+			X1(Parachain(u32::from(ParachainInfo::parachain_id()))),
+		).into(),
+		tur_per_second()
+	);
+
+	pub TurCanonicalPerSecond: (AssetId, u128) = (
+		MultiLocation::new(
+			0,
+			Here,
+		).into(),
+		tur_per_second()
+	);
+
+	pub KsmPerSecond: (AssetId, u128) = (MultiLocation::parent().into(), ksm_per_second());
+
+	pub KarPerSecond: (AssetId, u128) = (
+		MultiLocation::new(
+			1,
+			X2(Parachain(parachains::karura::ID), GeneralKey(parachains::karura::KAR_KEY.to_vec())),
+		).into(),
+		//KAR:KSM 50:1
+		ksm_per_second() * 50
+	);
+
+	pub KusdPerSecond: (AssetId, u128) = (
+		MultiLocation::new(
+			1,
+			X2(Parachain(parachains::karura::ID), GeneralKey(parachains::karura::KUSD_KEY.to_vec())),
+		).into(),
+		// kUSD:KSM = 400:1
+		ksm_per_second() * 400
+	);
+
+	pub LksmPerSecond: (AssetId, u128) = (
+		MultiLocation::new(
+			1,
+			X2(Parachain(parachains::karura::ID), GeneralKey(parachains::karura::LKSM_KEY.to_vec())),
+		).into(),
+		// LKSM:KSM = 10:1
+		ksm_per_second() * 10
+	);
+}
+
+pub struct ToTreasury;
+impl TakeRevenue for ToTreasury {
+    fn take_revenue(revenue: MultiAsset) {
+        if let MultiAsset {
+            id: AssetId::Concrete(id),
+            fun: Fungibility::Fungible(amount),
+        } = revenue
+        {
+            if let Some(currency_id) = CurrencyIdConvert::convert(id) {
+                // Ensure TreasuryAccount have ed requirement for native asset, but don't need
+				// ed requirement for cross-chain asset because it's one of whitelist accounts.
+				// Ignore the result.
+				let _ = Currencies::deposit(currency_id, &TreasuryAccount::get(), amount);
```

Comment:
nit: formatting.

Is it worth logging or something if this fails here? This seems like the sort of thing that we wouldn't just want to fail silently.

Reply:
I need to think about how to handle these sorts of things. Will add later this week